packages/interactive_media_ads/lib/src/android/android_ad_display_container.dart

```diff
@@ -0,0 +1,241 @@
+    @visibleForTesting InteractiveMediaAdsProxy? imaProxy,
+    @visibleForTesting PlatformViewsServiceProxy? platformViewsProxy,
+  }) {
+    return AndroidAdDisplayContainerCreationParams(
+      key: params.key,
+      onContainerAdded: params.onContainerAdded,
+      imaProxy: imaProxy,
+      platformViewsProxy: platformViewsProxy,
+    );
+  }
+
+  final InteractiveMediaAdsProxy _imaProxy;
+  final PlatformViewsServiceProxy _platformViewsProxy;
+}
+
+/// Android implementation of [PlatformAdDisplayContainer].
+base class AndroidAdDisplayContainer extends PlatformAdDisplayContainer {
+  /// Constructs an [AndroidAdDisplayContainer].
+  AndroidAdDisplayContainer(super.params) : super.implementation() {
+    final WeakReference<AndroidAdDisplayContainer> weakThis =
+        WeakReference<AndroidAdDisplayContainer>(this);
+    _videoView = _setUpVideoView(weakThis);
+    _frameLayout.addView(_videoView);
+    _videoAdPlayer = _setUpVideoAdPlayer(weakThis);
+  }
+
+  // The duration between each update to the IMA SDK of the progress of the
+  // currently playing ad.
+  static const int _progressPollingMs = 250;
+
+  // ViewGroup used to create the `ima.AdDisplayContainer`.
+  late final ima.FrameLayout _frameLayout =
+      _androidParams._imaProxy.newFrameLayout();
+
+  // Handles ad playback.
+  late final ima.VideoView _videoView;
+  ima.MediaPlayer? _mediaPlayer;
+
+  /// Callbacks that update the state of ad playback.
+  @internal
+  final Set<ima.VideoAdPlayerCallback> videoAdPlayerCallbacks =
+      <ima.VideoAdPlayerCallback>{};
+
+  // Handles ad playback callbacks from the IMA SDK.
+  late final ima.VideoAdPlayer _videoAdPlayer;
+
+  /// The native Android AdDisplayContainer.
+  @internal
+  ima.AdDisplayContainer? adDisplayContainer;
+
+  // Currently loaded ad.
+  ima.AdMediaInfo? _loadedAdMediaInfo;
+
+  // The saved ad position, used to resume ad playback following an ad
+  // click-through.
+  int _savedAdPosition = 0;
+
+  // Timer used to periodically update the IMA SDK of the progress of the
+  // currently playing ad.
+  Timer? _adProgressTimer;
+
+  int? _adDuration;
+
+  late final AndroidAdDisplayContainerCreationParams _androidParams =
+      params is AndroidAdDisplayContainerCreationParams
+          ? params as AndroidAdDisplayContainerCreationParams
+          : AndroidAdDisplayContainerCreationParams
+              .fromPlatformAdDisplayContainerCreationParams(params);
+
+  @override
+  Widget build(BuildContext context) {
+    return AndroidViewWidget(
+      key: params.key,
+      view: _frameLayout,
+      platformViewsServiceProxy: _androidParams._platformViewsProxy,
+      onPlatformViewCreated: () async {
+        adDisplayContainer = await _androidParams._imaProxy
+            .createAdDisplayContainerImaSdkFactory(
+          _frameLayout,
+          _videoAdPlayer,
+        );
+        params.onContainerAdded(this);
+      },
+    );
+  }
+
+  void _resetPlayer() {
```

Comment:
This could use a comment about intended use, since it only resets a subset of the state.